sql-plugin/src/main/java/com/nvidia/spark/rapids/ColumnarCopyHelper.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids;
+
+import ai.rapids.cudf.HostColumnVector.ColumnBuilder;
+
+import org.apache.spark.sql.vectorized.ColumnVector;
+
+/**
+ * A helper class which efficiently transfers different types of host columnar data into cuDF.
```

Comment:
it would be nice to explain this more and mention specifically not use to scala for loops.

Reply:
Note that using Scala primitive loops is fine
```
var i = 0
while (i < n) {
   /
   i = i + 1
}
```
Performance degradation comes from Boxing/Unboxing ints in the for comprehension since for comprehensions are syntactic sugar for function application to each element in a range.

See https://stackoverflow.com/questions/21373514/why-are-scala-for-loops-slower-than-logically-identical-while-loops

Reply:
I put the explanation into the comment. And I tested with Scala while-loops. Its performance was roughly same as Java loops.